gymnasium/dev_wrappers/lambda_action.py

```diff
@@ -28,3 +35,152 @@ def __init__(
+
+    Composite action space example:
+        >>> env = ExampleEnv(action_space=Dict(body=Dict(head=Box(0.0, 10.0, (1,))), left_arm=Discrete(4), right_arm=Box(0.0, 5.0, (1,))))
+        >>> env.action_space
+        Dict(body: Dict(head: Box(0.0, 10.0, (1,), float32)), left_arm: Discrete(4), right_arm: Box(0.0, 5.0, (1,), float32))
+        >>> args = {"right_arm": (0, 2), "body": {"head": (0, 3)}}
+        >>> env = ClipActionV0(env, args)
+        >>> env.action_space
+        Dict(body: Dict(head: Box(0.0, 3.0, (1,), float32)), left_arm: Discrete(4), right_arm: Box(0.0, 2.0, (1,), float32))
+    """
+
+    def __init__(self, env: gym.Env, args: CompositeParameterType):
+        """Constructor for the clip action wrapper.
+
+        Args:
+            env (Env): The environment to wrap
+            args (TreeParameterType): The arguments for clipping the action space
+        """
+        action_space = self._transform_space(env, args)
+
+        super().__init__(
+            env, lambda action, args: jp.clip(action, *args), args, action_space
+        )
+
+
+class RescaleActionsV0(LambdaCompositeActionV0):
+    """A wrapper that scales actions passed to :meth:`step` with a scale factor.
+
+    Basic Example:
+        >>> import gymnasium
+        >>> from gymnasium.wrappers import RescaleActionsV0
+        >>> env = gym.make('BipedalWalker-v3')
+        >>> env.action_space
+        Box(-1.0, 1.0, (4,), float32)
+        >>> env = RescaleActionsV0(env, (-0.5, 0.5))
+        >>> env.action_space
+        Box(-0.5, 0.5, (4,), float32)
+
+    Composite action space example:
+        >>> env = ExampleEnv(
+        ...    action_space=Dict(left_arm=Box(-2, 2, (1,)), right_arm=Box(-2, 2, (1,))
+        ... )
+        >>> env = RescaleActionsV0(env, {"left_arm": (-1,1), "right_arm": (-1,1)})
+        >>> env.action_space
+        Dict(left_arm: Box(-1, 1, (1,), float32), right_arm: Box(-1, 1, (1,), float32))
+    """
+
+    def __init__(self, env: gym.Env, args: CompositeParameterType):
+        """Constructor for the scale action wrapper.
+
+        Args:
+            env (Env): The environment to wrap
+            args (TreeParameterType): The arguments for scaling the actions
+        """
+        action_space = self._transform_space(env, args)
+        args = self._make_scale_args(env, args)
+
+        super().__init__(env, self._scale, args, action_space)
+
+    @staticmethod
+    def _scale(action: ActType, args: CompositeParameterType) -> jp.ndarray:
+        new_low, new_high = args[:2]
+        old_low, old_high = args[2:]
+
+        return jp.clip(
+            old_low
+            + (old_high - old_low) * ((action - new_low) / (new_high - new_low)),
+            old_low,
+            old_high,
+        )
+
+    @staticmethod
+    def _make_scale_args(
+        env: gym.Env, args: CompositeParameterType
+    ) -> CompositeParameterType:
+        return make_scale_args(env.action_space, args, make_scale_args)
```

Comment:
What's happening here and why? Why does the function need to take itself as an argument?